code/game/objects/items/stacks/sheets/leather.dm

```diff
@@ -73,50 +77,56 @@
+/obj/item/stack/material/animalhide/barehide
+	name = "bare hide"
+	desc = "This hide was stripped, but still needs to be tanned."
+	singular_name = "bare hide piece"
 	icon_state = "sheet-hairlesshide"
+	default_type = "bare hide"
+	bare = TRUE
 
-/obj/item/stack/material/wetleather
+/obj/item/stack/material/animalhide/wetleather
 	name = "wet leather"
 	desc = "This leather has been cleaned but still needs to be dried."
 	singular_name = "wet leather piece"
 	icon_state = "sheet-wetleather"
+	default_type = "wet leather"
 	var/wetness = 30 //Reduced when exposed to high temperautres
 	var/drying_threshold_temperature = 500 //Kelvin to start drying
 	icon_has_variants = TRUE
+	bare = TRUE
 
 //Step one - dehairing.
-/obj/item/stack/material/animalhide/attackby(obj/item/W as obj, mob/user as mob)
-	if(	istype(W, /obj/item/material/knife) || \
-		istype(W, /obj/item/material/kitchen/utensil/knife) || \
-		istype(W, /obj/item/material/twohanded/fireaxe) || \
-		istype(W, /obj/item/material/hatchet) )
+/obj/item/stack/material/animalhide/attackby(obj/item/W, mob/user)
+	if(W.sharp && !W.noslice && !bare && !W.iswirecutter()) //Can we cut and slice with the item? And does the hide still have something to remove? Say no to wirecutters since it's more about bladed items.
 
 		//visible message on mobs is defined as visible_message(var/message, var/self_message, var/blind_message)
-		usr.visible_message("<span class='notice'>\The [usr] starts cutting hair off \the [src]</span>", "<span class='notice'>You start cutting the hair off \the [src]</span>", "You hear the sound of a knife rubbing against flesh")
+		user.visible_message(span("notice", "\The [user] starts slicing the [hide_type] from \the [src]."),
+				span("notice", "You start slicing the [hide_type] from \the [src]"),
+				span("notice", "You hear the sound of a knife rubbing against flesh."))
 		if(do_after(user,50))
-			to_chat(usr, "<span class='notice'>You cut the hair from this [src.singular_name]</span>")
-			//Try locating an exisitng stack on the tile and add to there if possible
-			for(var/obj/item/stack/material/hairlesshide/HS in usr.loc)
-				if(HS.amount < 50)
-					HS.amount++
-					src.use(1)
-					break
-			//If it gets to here it means it did not find a suitable stack on the tile.
-			var/obj/item/stack/material/hairlesshide/HS = new(usr.loc)
-			HS.amount = 1
+			if(src.amount <= 0) //if we used it all, we don't get extra bare hide, no matter how much we clicked. Checks before we remove from the stack.
+				return
 			src.use(1)
+			to_chat(user, span("notice", "You remove the [hide_type] from the [singular_name]."))
+			playsound(user.loc, drop_sound, 50, 1)
+
+			//Try locating an exisitng stack on the tile and add to there if possible
+			for(var/obj/item/stack/material/animalhide/barehide/BH in user.loc)
+				if(BH.amount < 50)
+					BH.amount ++
+					to_chat(user, "You add the newly-stripped hide to the stack. It now contains [BH.amount] hides.")
+					return

```

Comment:
This chain seems malignant. Why use the loop if we're going to return immediately?

Reply:
because I'm a fool who was just trying to edit ancient code that was already there. Will get on that.